File: Kisan_connect/KisaanConnect/src/main/java/com/example/demo/entity/FarmerProduct.java
package com.example.demo.entity;

import java.math.BigDecimal;
import java.util.List;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "farmer_product")
public class FarmerProduct {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer fpid;

    private Integer stock;
    private BigDecimal price;

    @ManyToOne
    @JoinColumn(name = "sid")
    @JsonIgnoreProperties({"farmerProducts", "user"}) // Avoid recursion with farmer and user details
    private Farmer farmer;

    @ManyToOne
    @JoinColumn(name = "pid")
    @JsonIgnoreProperties("farmerProducts") // Avoid recursion with product details
    private Product product;

    @ManyToOne
    @JoinColumn(name = "vid")
    @JsonIgnoreProperties("farmerProducts") // Avoid recursion with variety details
    private Variety variety;

    @OneToMany(mappedBy = "farmerProduct")
    @JsonIgnoreProperties("farmerProduct") // Avoid recursion with cart and order details
    private List<Cart> carts;

    @OneToMany(mappedBy = "farmerProduct")
    @JsonIgnoreProperties("farmerProduct") // Avoid recursion with cart and order details
    private List<Order> orders;
}

File: DotNetLoginService/UserLoginService/Models/FarmerProduct.cs
using System;
using System.Collections.Generic;

namespace UserLoginService.Models
{
    public partial class FarmerProduct
    {
        public FarmerProduct()
        {
            Carts = new HashSet<Cart>();
            Orders = new HashSet<Order>();
        }

        public int Fpid { get; set; }
        public int? Sid { get; set; }
        public int? Pid { get; set; }
        public int? Stock { get; set; }
        public decimal? Price { get; set; }
        public int? Vid { get; set; }

        public virtual Product? PidNavigation { get; set; }
        public virtual Farmer? SidNavigation { get; set; }
        public virtual ICollection<Cart>? Carts { get; set; }
        public virtual ICollection<Order>? Orders { get; set; }
    }
}
